warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -491,7 +491,7 @@ private void handleException(Exception e) throws IOException {
             tce = (TabletClosedException) reason;
         
         int depth = 1;
-        while (tce == null && iie == null && ioe == null && reason.getCause() != null && reason.getCause() != reason && depth < 100) {
+        while (iie == null && reason.getCause() != null && reason.getCause() != reason && depth < 100) {
```

Comment:
I am wondering if we should even remove the iie==null check?

Reply:
if we reach an iie we always will be re-running so there is no reason to keep looking